assets/0022-halight-ln-eth-swap.puml

```diff
@@ -0,0 +1,90 @@
+@startuml
+
+title HALight-HAN atomic swap
+
+actor Alice as a
+
+participant "Alice's CND" as ac
+participant "Alice's LND" as al
+
+participant "Ethereum Chain" as eth
+
+participant "Bob's LND" as bl
+participant "Bob's CND" as bc
+
+actor Bob as b
+
+... We assume a negotiation and communication protocol is in place.\n \
+Both parties have agreed to the same set of parameters and are ready to swap. ...
+
+par
+
+b->bc: Get swap
+bc-->b: Action: lnd-add-hold-invoice
+b->bl: Add hold invoice
+
+a->ac: Get swap
+ac-->a: Action: lnd-pay-invoice
+a->al: Pay invoice
+
+note right a
+You might notice a potential race-condition here. (We are in a par block!)
+What if Alice tries to pay the invoice before Bob added it do his LND node?
+
+The answer to this is "try again". We already make an assumption that cnd's
+clients can successfully make transactions.
+Failing to make a payment because the invoice does not yet exist is similar
+to failing to make a Bitcoin transaction because the fee was to low.
```

Comment:
If the fee was too low the transaction would remain pending for a long time (some nodes never clear the mempool).

```suggestion
to failing to make a Bitcoin transaction because it was invalid.
```